sys/net/link_layer/l2util/l2util.c

```diff
@@ -40,29 +40,20 @@ int l2util_eui64_from_addr(int dev_type, const uint8_t *addr, size_t addr_len,
                            eui64_t *eui64)
 {
     switch (dev_type) {
-#if defined(MODULE_NETDEV_ETH) || defined(MODULE_ESP_NOW)
+#if defined(MODULE_NETDEV_ETH) || defined(MODULE_ESP_NOW) || \
+    defined(MODULE_NORDIC_SOFTDEVICE_BLE) || defined(MODULE_NIMBLE_NETIF)
         case NETDEV_TYPE_ETHERNET:
         case NETDEV_TYPE_ESP_NOW:
+        case NETDEV_TYPE_BLE:
             if (addr_len == sizeof(eui48_t)) {
                 eui48_to_eui64(eui64, (const eui48_t *)addr);
                 return sizeof(eui64_t);
             }
             else {
                 return -EINVAL;
             }
-#endif  /* defined(MODULE_NETDEV_ETH) || defined(MODULE_ESP_NOW) */
-#if defined(MODULE_NORDIC_SOFTDEVICE_BLE) || defined(MODULE_NIMBLE_NETIF)
-        case NETDEV_TYPE_BLE:
-            eui64->uint8[0] = addr[0];
-            eui64->uint8[1] = addr[1];
-            eui64->uint8[2] = addr[2];
-            eui64->uint8[3] = 0xff;
-            eui64->uint8[4] = 0xfe;
-            eui64->uint8[5] = addr[3];
-            eui64->uint8[6] = addr[4];
-            eui64->uint8[7] = addr[5];
-            return sizeof(eui64_t);
-#endif  /* defined(MODULE_NORDIC_SOFTDEVICE_BLE) || defined(MODULE_NIMBLE_NETIF) */
```

Comment:
Note for posterity: We discovered offline that this does exactly the same as `eui48_to_eui64()` so code was unnecessarily duplicated here.